### PR TITLE
realvuAnalyticsAdapter

### PR DESCRIPTION
**Pull request by tchibirev**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
RealVu Analytics Adapter (realvuAnalyticsAdapter.js)

- test parameters for validating bids
```
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
it@realvu.com
- [x] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

## Other information

Resubmission of PR #1640

### CODE REVIEW

**Comment by mkendall07** [comment]
@tchibirev 
looks much better, thank you!

**Comment by tchibirev** [comment]
Matt,
Yes, much easier. But I still see
Cannot start ChromeHeadless in Travis CI (as any other PRs)
I hope you will handle it with higher priority because of it is a resubmission of PR #1640
Thanks,
Igor




On Jan 16, 2018, at 2:50 PM, Matt Kendall <notifications@github.com<mailto:notifications@github.com>> wrote:


@tchibirev<https://github.com/tchibirev>
looks much better, thank you!

**Comment by mkendall07** [comment]
Igor, yep working on a fix for travis.

File: modules/realvuAnalyticsAdapter.js [hunk 1]
```diff
@@ -0,0 +1,975 @@
...
+        if (bk.a.rnd < Math.pow(10, 1 - (z.sr.charCodeAt(0) & 7))) {
+          z.scr(bk.s1, bk.a);
+        }
+        bk = z.beacons.shift();
+      }
+    }
+  },
+
+  scr: function (s1, a) {
+    var st = document.createElement('script');
```
Comment (mkendall07) [inline modules/realvuAnalyticsAdapter.js]: really shouldn't be downloading additional javascript after the module is loaded. What's the use case here? The analytics file is already very large.

File: modules/realvuAnalyticsAdapter.js [hunk 2]
```diff
@@ -0,0 +1,975 @@
...
+      // if(window.top1.RealVu_test)window.top1.RealVu_log.write("FindPosG Exception: "+e.message+ " xp="+xp+" yp="+yp );
+    }
+    var q = {
+      'x': Math.round(xp),
+      'y': Math.round(yp)
+    };
+    return q;
+  },
+
+  poll: function () {
```
Comment (mkendall07) [inline modules/realvuAnalyticsAdapter.js]: what's the polling for? Generally a bad practice.

File: modules/realvuAnalyticsAdapter.js [hunk 3]
```diff
@@ -0,0 +1,975 @@
...
+
+const utils = require('src/utils.js');
+
+var realvuAnalyticsAdapter = adapter({
+  global: 'realvuAnalytics',
+  handler: 'on',
+  analyticsType: 'bundle'
+});
+/*
+  Copyright(C) 2001-2018, RealVu Inc., All rights reserved. Use of RealVu’s patented, patent pending,
```
Comment (mkendall07) [inline modules/realvuAnalyticsAdapter.js]: code submitted to the project is governed by the prebid license. You cannot add a separate license here.

File: modules/realvuAnalyticsAdapter.js [hunk 4]
```diff
@@ -0,0 +1,975 @@
...
+  copyrighted and proprietary code is not being made available under an Apache or other open source license,
+  but is available only under a proprietary license of RealVu Inc.
+*/
+window.top1 = window;
```
Comment (mkendall07) [inline modules/realvuAnalyticsAdapter.js]: this is also considered a bad practice, polluting the global namespace.

File: modules/realvuAnalyticsAdapter.js (hunk 3 above)
Comment (markhe) [reply modules/realvuAnalyticsAdapter.js]: @mkendall07 ,  RealVu has looked into this with our legal counsel and believe that it is acceptable, appropriate and necessary to include this copyright protection notice in the adapter.  Our objective is to protect our intellectual property.  We are not contributing our proprietary code to the upstream project.  This notice excludes it from the Apache license.  If you believe it is still unacceptable, could you please let us know who we can contact at Prebid.org to further this discussion?
Thx,
-mark

Comment (mkendall07) [reply modules/realvuAnalyticsAdapter.js]: @markhe 
The recommended approach here would be to put your proprietary code into an external js file and load it from the analytics adapter. You may include the license on that file. Thanks

**Comment by mkendall07** [comment]
Closing per our recommendation to use external script. Thanks

**Pull request closed without merging**